cterasdk/core/plans.py

```diff
@@ -23,3 +37,82 @@ def get(self, name, include=None):
         if plan.name is None:
             raise CTERAException('Could not find subscription plan', None, name=name)
         return plan
+
+    def add(self, name, retention=None, quotas=None):
+        """
+        Add a subscription plan
+
+        :param dict,optional quotas: The data retention policy
+        :param dict,optional quotas: The items included in the plan and their respective quota
+        """
+        plan = defaults.default_class(self._portal, 'Plan')
+        plan.name = name
+        Plans._assign_retention(plan, retention)
+        Plans._assign_quotas(plan, quotas)
+        try:
+            response = self._portal.add('/plans', plan)
+            logging.getLogger().info("Plan created. %s", {'plan': name})
+            return response
+        except CTERAException as error:
+            logging.getLogger().error("Plan creation failed.")
+            raise CTERAException('Plan creation failed', error)
+
+    def modify(self, name, retention=None, quotas=None):
+        """
+        Modify a subscription plan
+
+        :param dict,optional quotas: The data retention policy
+        :param dict,optional quotas: The items included in the plan and their respective quota
+        """
+        plan = self.get(name)
+        Plans._assign_retention(plan, retention)
+        Plans._assign_quotas(plan, quotas)
+        print(plan)
+        input()
+        try:
+            response = self._portal.put('/plans/' + name, plan)
+            logging.getLogger().info("Plan modified. %s", {'plan': name})
+            return response
+        except CTERAException as error:
+            logging.getLogger().error("Could not modify subscription plan.")
+            raise CTERAException('Could not modify subscription plan', error)
+
+    @staticmethod
+    def _assign_retention(plan, retention):
+        if isinstance(retention, dict):
```

Comment:
I'm guessing the idea was to check if it's not None, right?

Reply:
Correct. I preferred to check it's of type dict in order not to fail on the ".get" commands inside the if block